Survey/Sources/Presentation/Modules/Home/HomeViewModel.swift

```diff
@@ -9,23 +9,84 @@
 import Combine
 import SwiftUI
 
-struct HomeViewModel {}
+struct HomeViewModel {
+
+    let useCase: HomeUseCaseProtocol
+}
 
 extension HomeViewModel: ViewModel {
 
     func transform(_ input: Input) -> Output {
-        return Output()
+        let errorTracker = ErrorTracker()
+        let activityTracker = ActivityTracker(false)
+        var pageNumber = 0
+        let output = Output()
+
+        input.loadTrigger
+            .map { _ in
+                self.useCase.getSurveyList(pageNumber: pageNumber + 1, pageSize: 10)
+                    .trackError(errorTracker)
+                    .trackActivity(activityTracker)
+                    .asDriver()
+            }
+            .switchToLatest()
+            .map {
+                pageNumber += 1
+                if !UserStorage.cachedSurveyList.hasSameChildren(as: $0 as? [APISurvey] ?? []) {
```

Comment:
Would you mind explaining to me a little bit about this **hasSameChildren**?

Reply:
I used `hasSameChildren` to check whether there are any changes between **cachedSurveyList** (cached from opening app) and **surveys list** (got from API response), if there is, then up date the cache and propagated to Home Screen. If there is no, we keep current survey list that's displayed on home screen.

Inside `hasSameChildren`, we compare them after sorting them and making sure they are the same number of items.
 ```
   func hasSameChildren(as other: [Element]) -> Bool {
        return count == other.count && sorted() == other.sorted()
    }
```
As explained above, is it clear to you or if it didn't answer your concern correctly, feel free to ask me more 😄 . Thank youu.

Reply:
Yes, I understood the part where we compare the count but this is only for the idea case that data never changes. But it's fine for the IC. Thank for your explanation